boot/freeldr/freeldr/arch/i386/multiboot.S

```diff
@@ -91,6 +95,36 @@ MultibootEntry:
     cmp eax, MULTIBOOT_BOOTLOADER_MAGIC
     jne mbfail
 
+    /* Save multiboot info structure */
+    mov esi, ebx
+    mov edi, offset MultibootInfo + INITIAL_BASE - FREELDR_BASE
+    mov ecx, MB_INFO_SIZE
+    shr ecx, 2
+    rep movsd
+    mov dword ptr ds:[MultibootInfo + INITIAL_BASE - FREELDR_BASE + MB_INFO_MMAP_ADDR_OFFSET], 0
+    mov dword ptr ds:[_MultibootInfoPtr + INITIAL_BASE - FREELDR_BASE], offset MultibootInfo
```

Comment:
I think that we can remove the MultibootInfoPtr variable, since we "know" where the MultibootInfo is: it is in MultibootInfo ! (so that in C code you could use something like &MultibootInfo ).

Reply:
`&MultibootInfo` will be always defined, and `MultibootInfoPtr` will be NULL if we're booting non-multiboot way, it's more proper way I think.

Reply:
ok you've a good point there.